### PR TITLE
Bump both tracing and tracing-effectful to 1.0.0.0

### PR DESCRIPTION
**Pull request by Raveline**
Let's use proper versioning.

### CODE REVIEW

File: tracing-effectful/tracing-effectful.cabal [hunk 1]
```diff
@@ -33,7 +33,7 @@ library
     , base            <5
     , effectful-core  >=1.0.0.0 && <3.0.0.0
     , exceptions      <0.11
-    , tracing         >=0.0.7   && <1.0.0.0
+    , tracing         <=1.0.0.0
```
Comment (jonathanjouty) [inline tracing-effectful/tracing-effectful.cabal:36]: _Nitpick:_ no upper bound?

File: tracing-effectful/tracing-effectful.cabal [hunk 2]
```diff
@@ -1,6 +1,6 @@
 cabal-version:   3.0
 name:            tracing-effectful
-version:         0.1.0.0
+version:         1.0.0.0
```
Comment (jonathanjouty) [inline tracing-effectful/tracing-effectful.cabal:3]: We didn't need to a major-major bump here, is the idea to keep this version in sync with `tracing`?
If so, we should add a comment about that.

File: tracing-effectful/tracing-effectful.cabal (hunk 1 above)
Comment (Raveline) [reply tracing-effectful/tracing-effectful.cabal:36]: You mean lower ? (To be honest, it's mostly to satisfy the CI, it should probably `==` here)

File: tracing-effectful/tracing-effectful.cabal (hunk 2 above)
Comment (Raveline) [reply tracing-effectful/tracing-effectful.cabal:3]: In sync would be a bit much, it was more for the sake of parallelism (I find it be a bit weird to have `ŧracing` be 1.0.0.0 and `tracing-effectful` be 0.1.0.0);

File: tracing-effectful/tracing-effectful.cabal (hunk 1 above)
Comment (jonathanjouty) [reply tracing-effectful/tracing-effectful.cabal]: 🙈 Yes :)
In that case it's just fine
BTW Cabal now also allows `1.0.*` I think

Comment (Raveline) [reply tracing-effectful/tracing-effectful.cabal]: I've put `==` (and the bound checker of the CI seems ok with it :relieved:), as this is directly linked to the library, I think we should use an exact version for now.

**Review by jonathanjouty — APPROVED** [review]

**Pull request merged**